moon/artifacts.py

```diff
@@ -114,52 +86,93 @@ def run(self):
                 while timestamp <= now:
                     timestamp = self.waitForImage()
                     dropped += 1
-                self.detect(self.left_image, self.right_image)
+                self.detect_and_publish(self.left_image, self.right_image)
         except BusShutdownException:
             pass
 
+    def detect_and_publish(self, left_image, right_image):
+        results = self.detect(left_image, right_image)
+        for r in results:
+            self.publish('artf', r)
+
     def detect(self, left_image, right_image):
-        limg = cv2.imdecode(np.fromstring(left_image, dtype=np.uint8), 1)
-        rimg = cv2.imdecode(np.fromstring(right_image, dtype=np.uint8), 1)
+        results = []
 
+        limg = cv2.imdecode(np.frombuffer(left_image, dtype=np.uint8), 1)
```

Comment:
maybe instead of 1 use `cv2.IMREAD_COLOR` as suggested in https://github.com/robotika/osgar/pull/495#discussion_r427808538

Reply:
ok